File: skiresort.cpp
#include <iostream>
using ll = long long;
using namespace std;

long long num_substrings(long long length,long long minimum){
    long long ans = 0;
    if (minimum > length){
        return 0;
    }
    for (long long substring = minimum; substring < length+1; substring++){
        //printf("%d %d %d \n",length,substring,length-(substring-1));
        ans += length-(substring-1);
    }
    return ans;
}

int main(){
    long long testcases,n,k;
    long long temp;
    long long q;
    cin >> testcases;

    for (long long t = 0; t < testcases; t++){
        cin >> n >> k >> q;
        long long ans = 0;
        long long consecutive = 0;
        for (long long x = 0; x < n; x++){
            cin >> temp;
            //printf("TEMP: %d \n",temp);
            if (temp <= q){
                consecutive++;
            } else {
                //printf("CONSECUTIVE %d \n",consecutive);
                ans += num_substrings(consecutive,k);
                consecutive = 0;
            }
        }
        if (consecutive != 0){
            ans += num_substrings(consecutive,k);
        }
        printf("%d \n",ans);
    }
}

File: skiresort.py
def len_substrings(string_len,min_substring_len):
    ans = 0
    for substring_len in range(min_substring_len,string_len+1):
        #print(string_len,substring_len,string_len-(substring_len-1))
        ans += string_len-(substring_len-1)
    return ans

print(len_substrings(3,1))

testcases = int(input())
for i in range(testcases):
    n,k,q = map(int,input().split())
    len_holiday = {i:0 for i in range(-1,n)}
    temps = list(map(int,input().split()))
    for x in range(n):
        if temps[x] <= q:
            len_holiday[x] = len_holiday[x-1] + 1
    values = list(len_holiday.values())
    #print(values)
    vals = [values[i] for i in range(len(values)) if values[i] != 0 and (i == len(values)-1 or values[i+1] == 0)]
    #print(vals)
    ans = 0
    for consecutive in vals:
        ans += len_substrings(consecutive,k)
    print(ans)